confirmation_rule.py

```diff
@@ -397,79 +403,68 @@ def will_checkpoint_be_justified(store: Store, checkpoint: Checkpoint) -> bool:
-            
-            # We can only rely on the previous head if it is a descendant of the block we are attempting to confirm
-            if not is_ancestor(store, store.prev_slot_head, block_root):
+    canonical_roots = get_chain_roots(store, latest_confirmed_root, head)
+    lmd_confirmed_root = latest_confirmed_root
+
+    for block_root in canonical_roots:
+        block_epoch = get_block_epoch(store, block_root)
+        if block_epoch > get_block_epoch(store, lmd_confirmed_root):
+            # Block won't get filtered out in the next epoch
+            if not will_canonical_block_match_filtering_criteria(store, block_root):
                 break
-            
-            if not is_one_confirmed(store, block_root):
-                break
-            
-            confirmed_root = block_root
-            
-    if (is_first_slot_in_epoch(get_current_slot(store))
-        or store.unrealized_justifications[head].epoch + 1 >= current_epoch):
-        # get suffix of the canonical chain
-        canonical_roots = get_chain_roots(store, confirmed_root, head)
-
-        tentative_confirmed_root = confirmed_root
-
-        for block_root in canonical_roots:
-            block_epoch = get_block_epoch(store, block_root)
-            tentative_confirmed_epoch = get_block_epoch(store, tentative_confirmed_root)
-            
-            # The following condition can only be true the first time that we advance to a block from the current epoch
-            if block_epoch > tentative_confirmed_epoch:
-                # To confirm blocks from the current epoch ensure that
-                # current epoch checkpoint will be justified
-                checkpoint = get_checkpoint_for_block(store, block_root, block_epoch)
-                if not will_checkpoint_be_justified(store, checkpoint):
-                    break  
-
-            if not is_one_confirmed(store, block_root):
+
+            # Current epoch checkpoint will be justified
+            checkpoint = get_checkpoint_for_block(store, block_root, block_epoch)
+            if not will_checkpoint_be_justified(store, checkpoint):
                 break
-                
-            tentative_confirmed_root = block_root
-            
-        # the tentative_confirmed_root can only be confirmed if we can ensure that it is not going to be reorged out in either the current or next epoch.
-        if (get_block_epoch(store, tentative_confirmed_root) == current_epoch
-            or (get_voting_source(store, tentative_confirmed_root).epoch + 2 >= current_epoch
-                and (is_first_slot_in_epoch(get_current_slot(store))
-                     or will_no_conflicting_checkpoint_be_justified(store, get_checkpoint_for_block(store, head, current_epoch))))):
-            confirmed_root = tentative_confirmed_root
-            
-    return confirmed_root        
+
+        if is_one_confirmed(store, block_root):
+            lmd_confirmed_root = block_root
+        else:
+            break
```

Comment:
Just a very cosmetic suggestion to make the statement consistent with other loop breaks 

```suggestion
        if not is_one_confirmed(store, block_root):
            break
        lmd_confirmed_root = block_root
```